src/matrixroomhandler.ts

```diff
@@ -74,19 +96,157 @@ export class MatrixRoomHandler {
       this.HandleInvite(event);
     } else if (event.type === "m.room.redaction" && context.rooms.remote) {
       this.discord.ProcessMatrixRedact(event);
-    } else if (event.type === "m.room.message" && context.rooms.remote) {
+    } else if (event.type === "m.room.message") {
       log.verbose("MatrixRoomHandler", "Got m.room.message event");
-      const srvChanPair = context.rooms.remote.roomId.substr("_discord".length).split("_", ROOM_NAME_PARTS);
-      return this.discord.ProcessMatrixMsgEvent(event, srvChanPair[0], srvChanPair[1]).catch((err) => {
-        log.warn("MatrixRoomHandler", "There was an error sending a matrix event", err);
-      });
+      if (event.content.body && event.content.body.startsWith("!discord")) {
+        return this.ProcessCommand(event, context);
+      } else if (context.rooms.remote) {
+        const srvChanPair = context.rooms.remote.roomId.substr("_discord".length).split("_", ROOM_NAME_PARTS);
+        return this.discord.ProcessMatrixMsgEvent(event, srvChanPair[0], srvChanPair[1]).catch((err) => {
+            log.warn("MatrixRoomHandler", "There was an error sending a matrix event", err);
+        });
+      }
     } else {
       log.verbose("MatrixRoomHandler", "Got non m.room.message event");
     }
   }
 
   public HandleInvite(event: any) {
-    // Do nothing yet.
+    log.info("MatrixRoomHandler", "Received invite for " + event.state_key + " in room " + event.room_id);
+    if (event.state_key === this.botUserId) {
+      this.bridge.getIntent().getClient().joinRoom(event.room_id).catch((err) => {
+        log.error("MatrixRoomHandler", err);
+        setTimeout(() => {
+            this.bridge.getIntent().join(event.room_id).catch((err2) => log.error("MatrixRoomHandler", err2));
```

Comment:
Need a bit more detail than just logging the error here. Especially since we are dropping to the floor after this.